### PR TITLE
Rewrote the storage file in ts

### LINKED ISSUE #37: Rewrite Storage.js file

### PR DESCRIPTION
**Pull request by imblowfish**
Closed #37

In this PR, the `Storage.js` file has been removed and a new `Storage.ts` has been added

I replaced the previously used initialization from the `App.tsx` file with isolated initialization from within the module, which in my opinion will reduce the number of potential errors and make API simpler

I also moved all test templates like empty array, full array, etc. to the `Storage.ts` class because I think those methods are more storage oriented.

### CODE REVIEW

File: src/data/Storage.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+function arrayOfCyclesWithSingleCycle(): Promise<void> {
+    return storage.set.cycles([
+        {
+        cycleLength: 28,
+        periodLength: 6,
+        startDate: "2023-05-31",
+        },
+    ] satisfies Cycle[]);
```
Comment (IraSoro) [inline src/data/Storage.ts:61]: I have a mistake: 'Cycle' only refers to a type, but is being used as a value her.
I fixed it when I "satisfies" it to "as"

File: src/data/Storage.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+function arrayOfCyclesWithSingleCycleWithDelay(): Promise<void> {
+    return storage.set.cycles([
+        {
+        cycleLength: 28,
+        periodLength: 6,
+        startDate: "2023-05-03",
+        },
+    ] satisfies Cycle[]);
```
Comment (IraSoro) [inline src/data/Storage.ts:71]: Here too

File: src/data/Storage.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        cycleLength: 26,
+        periodLength: 5,
+        startDate: "2023-04-07",
+        },
+        {
+        cycleLength: 25,
+        periodLength: 6,
+        startDate: "2023-03-13",
+        },
+    ] satisfies Cycle[]);
```
Comment (IraSoro) [inline src/data/Storage.ts:96]: Here too

File: src/data/Storage.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        cycleLength: 28,
+        periodLength: 5,
+        startDate: "2023-01-16",
+        },
+        {
+        cycleLength: 30,
+        periodLength: 6,
+        startDate: "2022-12-17",
+        },
+    ] satisfies Cycle[]);
```
Comment (IraSoro) [inline src/data/Storage.ts:136]: Here too

**Review by IraSoro — CHANGES_REQUESTED** [review]
I have a mistake: 'Cycle' only refers to a type, but is being used as a value her. 
I fixed it when I "satisfies" it to "as"

File: src/data/Storage.ts (hunk 1 above)
Comment (imblowfish) [reply src/data/Storage.ts:61]: I think you can update your `ts` package, because `satisfies` keyword was introduced in 4.9 version

**Review by IraSoro — APPROVED** [review]
